library/src/scala/tasty/reflect/TypeOrBoundsOps.scala

```diff
@@ -263,6 +271,14 @@ trait TypeOrBoundsOps extends Core {
     def cases(given ctx: Context): List[Type] = internal.MatchType_cases(self)
   }
 
+  /**
+   * An accessor for `scala.internal.MatchCase[_,_]`, the representation of a `MatchType` case.
+    */
+  def MatchCaseType(given Context): Type = {
+    import scala.internal.MatchCase
+    Type(classOf[MatchCase[_,_]])
```

Comment:
This is too hacky. In the future, we won't have access to the `classOf[MatchCase[_, _]]`. This should probably be in the `scala.internal` interface (`CompilerInterface`).

Reply:
This is interesting. Even in the tests, I had to use `Type(classOf[List[_]])` in order to get the type of `List`.

Actually writing `typeOf[List]` gives me `missing type parameter(s) for List`. Writing `typeOf[List[_]]` satisfies Dotty but gives me `AppliedType(List, TypeBounds(Nothing,Any))`, which is not helpful.

Is this a bug or am I doing it wrong?

Reply:
(leaving is as-is for now since it makes more sense to change this and the tests at the same time once we figure out the above question)